maven3-agent/src/main/java/org/jvnet/hudson/maven3/agent/Maven3Main.java

```diff
@@ -61,8 +61,22 @@
     private static Launcher launcher;
 
     public static void main(String... args) throws Exception {
-        main(new File(args[0]), new File(args[1]),new File(args[2]),
-             new File(args[3]), Integer.parseInt(args[4]));
+        String slaveAgentSocket = args[4];
+        int i = slaveAgentSocket.indexOf(':');
+        if (i > 0) {
+            main(new File(args[0]), new File(args[1]), new File(args[2]),
+                    new File(args[3]), slaveAgentSocket.substring(0, i), Integer.parseInt(slaveAgentSocket.substring(i+1)));
+        } else {
+            main(new File(args[0]), new File(args[1]), new File(args[2]),
+                    new File(args[3]), null, Integer.parseInt(slaveAgentSocket));
+        }
+    }
+
+    @Deprecated
+    public static void main(File m2Home, File remotingJar, File interceptorJar
+            , File interceptorCommonJar, int tcpPort) throws Exception {
```

Comment:
this is juste pre-existing code https://github.com/jenkinsci/maven-interceptors/pull/8/files#diff-728018f274162eddc91b43d449872129L84